src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,194 @@
+ *               : // violation, ':' should not be in separate line
+ *             break;
+ *     }
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code ws.colonInSeparateLine}
+ * </li>
+ * <li>
+ * {@code ws.preceded}
+ * </li>
+ * </ul>
+ *
+ * @since 8.42
+ */
+@StatelessCheck
+public class NoWhitespaceBeforeCaseDefaultColonCheck
+    extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "ws.preceded";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY_ON_SEPARATE = "ws.colonInSeparateLine";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.COLON};
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST parent = ast.getParent();
+        final boolean isColonOfCase = parent.getType() == TokenTypes.LITERAL_CASE;
+        final boolean isColonOfDefault = parent.getType() == TokenTypes.LITERAL_DEFAULT;
+        final boolean violation = isColonOfCase && checkWhitespaceBeforeColonOfCase(ast)
+                || isColonOfDefault && checkWhitespaceBeforeColonOfDefault(ast);
+        if (violation) {
+            log(ast, MSG_KEY, ast.getText());
+        }
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case.
+     *
+     * @param colonAst DetailAST to check.
+     * @return whether there is a whitespace before colonAst or not.
+     */
+    private boolean checkWhitespaceBeforeColonOfCase(DetailAST colonAst) {
+        final DetailAST previousSibling = colonAst.getPreviousSibling();
+        final boolean areOnSameLine = TokenUtil.areOnSameLine(previousSibling, colonAst);
+        boolean horizontalWhitespaceDetected = false;
+        if (areOnSameLine) {
+            int offset = 0;
+            if (previousSibling.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {
+                offset = 1;
+            }
+            horizontalWhitespaceDetected =
+                    colonAst.getColumnNo() != getLastColumnNumberOf(previousSibling) + offset;
+        }
+        else {
+            log(colonAst, MSG_KEY_ON_SEPARATE);
+        }
+        return horizontalWhitespaceDetected;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return whether there is a whitespace before colonAst or not.
```

Comment:
`@return true if there is whitespace preceding colon ast` this removes any ambiguity. Same for above javadoc.

Reply:
Done.

Reply:
Resolved.